File: iOS/ssabree/ssabree/Core/Repository/FakeAuthRepository.swift
import Foundation

final class FakeAuthRepository: AuthRepository {
    func login(email: String, password: String) async -> Result<Void, Error> { .success(()) }
    func refreshTokens() async -> Result<Void, Error> { .success(()) }
    func logout() async -> Result<Void, Error> { .success(()) }
    func isLoggedIn() -> Bool { true }
    func isEmailUsed(email: String) async -> Result<Bool, Error> { .success(false) }
    func signUp(info: SignUpInfo) async -> Result<Void, Error> { .success(()) }
    func requestSsafyVerification(mattermostId: String, generation: Int, name: String) async -> Result<Void, Error> { .success(()) }
    func confirmSsafyVerification(mattermostId: String, code: String) async -> Result<Void, Error> { .success(()) }
    func findId(mattermostId: String) async -> Result<String, Error> { .success("fake@example.com") }
    func resetPassword(mattermostId: String, newPassword: String) async -> Result<Void, Error> { .success(()) }
    func withdraw() async -> Result<Void, Error> { .success(()) }
    func getMyMemberId() async -> Result<Int, Error> { .success(1) }
}

File: android/ssabree/app/src/main/java/com/ssafy/ssabree/core/repository/test/FakeAuthRepository.kt
package com.ssafy.ssabree.core.repository.test

import com.ssafy.ssabree.core.repository.AuthRepository
import com.ssafy.ssabree.core.repository.model.SignUpInfo

class FakeAuthRepository: AuthRepository {
    override suspend fun login(email: String, password: String): Result<Unit> {
        return Result.success(Unit)
    }
    override suspend fun refreshTokens(): Result<Unit> {
        return Result.success(Unit)
    }
    override suspend fun logout(): Result<Unit> {
        return Result.success(Unit)
    }

    override suspend fun isEmailUsed(email: String): Result<Boolean> {
        return Result.success(false)
    }

    override suspend fun signUp(req: SignUpInfo): Result<Unit> {
        return Result.success(Unit)
    }

    override suspend fun requestSsafyVerification(mattermostId: String, generation: Int, name: String): Result<Unit> {
        return Result.success(Unit)
    }

    override suspend fun confirmSsafyVerification(
        mattermostId: String,
        code: String
    ): Result<Unit> {
        return Result.success(Unit)
    }

    override suspend fun requestPasswordResetCode(destination: String): Result<Unit> {
        return Result.success(Unit)
    }

    override suspend fun confirmPasswordReset(
        code: String,
        newPassword: String
    ): Result<Unit> {
        return Result.success(Unit)
    }

    override suspend fun findId(mattermostId: String): Result<String> {
        return Result.success("test@example.com")
    }

    override suspend fun resetPassword(mattermostId: String, newPassword: String): Result<Unit> {
        return Result.success(Unit)
    }

    override suspend fun withdraw(): Result<Unit> {
        return Result.success(Unit)
    }
}
